astroid/interpreter/_import/spec.py

```diff
@@ -451,23 +455,28 @@ def _find_spec(
     # Need a copy for not mutating the argument.
     modpath = list(module_path)
 
-    submodule_path = None
+    search_paths = None
     processed: list[str] = []
 
     while modpath:
         modname = modpath.pop(0)
+
+        submodule_path = search_paths or path
+        if submodule_path is not None:
+            submodule_path = tuple(submodule_path)
```

Comment:
Hey, thanks for the ping. My review won't take very long. One initial question that will save me a little time: can you mention under what circumstances this isn't already a tuple?